Import modules

dfunckt opened the pull request:
Merge a few dependencies together into one tree to provide complete coverage for the use case: given a docker-compose.yml, build the services, and (optionally) create a release for it at balenaCloud.

Clients that are looking to build docker-compose files typically have to depend on and import *all* of these modules, and some of the modules themselves depend on each other creating a situation where maintaining the correct versions of things is very cumbersome.

This commit is the first step to a standalone module that is reasonably useful but currently makes no effort to rethink the exposed APIs. Imported modules are exported as-is. In the future, we should improve by exporting more meaningful APIs.

The merged dependencies are:

- [balena-multibuild](https://github.com/balena-io-modules/balena-multibuild) @ aefcb3248ed183
- [balena-compose-parse](https://github.com/balena-io-modules/balena-compose-parse) @ 06a6b10d3096e
- [balena-release](https://github.com/balena-io-modules/balena-release) @ 83155848aa4c
- [docker-qemu-transpose](https://github.com/balena-io-modules/docker-qemu-transpose) @ 734d8397dfe
- [dockerfile-template](https://github.com/balena-io-modules/dockerfile-template) @ 3ae77d4ec75
- [resin-bundle-resolve](https://github.com/balena-io-modules/resin-bundle-resolve) @ 7a972ce731e5ec4
- [resin-docker-build](https://github.com/balena-io-modules/resin-docker-build) @ 703653cc0eaa57

dfunckt committed:
Import balena-compose-parse

Repo: https://github.com/balena-io-modules/balena-compose-parse
Commit: 06a6b10d3096e133762200e3737b67d3fe980ff7

Change-type: minor

dfunckt committed:
Import balena-release

Repo: https://github.com/balena-io-modules/balena-release
Commit: 83155848aa4cd6daac54647913f41aaebe3c1b3a

Change-type: minor

dfunckt committed:
Import balena-multibuild

Repo: https://github.com/balena-io-modules/balena-multibuild
Commit: aefcb3248ed183faa4019167bd4266e43ffa8d74

Change-type: minor

dfunckt committed:
Import resin-docker-build

Repo: https://github.com/balena-io-modules/resin-docker-build
Commit: 703653cc0eaa57eb50ff9a52adb3de126659de8f

Change-type: minor

dfunckt committed:
Import resin-bundle-resolve

Repo: https://github.com/balena-io-modules/resin-bundle-resolve
Commit: 7a972ce731e5ec40112869438b6586c87dd15108

Change-type: minor

dfunckt commented:
https://github.com/balena-io-modules/docker-qemu-transpose is another one that could be merged in this.

EDIT: Done.

dfunckt committed:
Import docker-qemu-transpose

Repo: https://github.com/balena-io-modules/docker-qemu-transpose
Commit: 734d8397dfe33ae3af85cdd4fb27c64a6ca77a25

Change-type: minor

toochevere commented on lib/multibuild/build.ts:
```diff
@@ -0,0 +1,346 @@
+	//   In order to support `platform`, all images used in the Dockerfile must use the same
+	//   architecture.  Determining this is problematic.  See comments in getRecommendedPlatformHandling
+
+	const usePlatformOption: boolean =
```
FYI:  Once this merges, in order to unblock the (formerly) multibuild issue, I'll PR a change to allow the default `--platform` behavior here with a new option on the build task `task.useDefaultPlatformForMultiarchBaseImages` and then 

```
	const usePlatformOption: boolean =
		!!task.dockerPlatform &&
		semver.satisfies(
			semver.coerce((await docker.version()).ApiVersion) || '0.0.0',
			'>=1.38.0',
		) &&
		(task.useDefaultPlatformForMultiarchBaseImages ||
			(await checkAllowDockerPlatformHandling(task, docker)));
```

dfunckt committed:
Import dockerfile-template

Repo: https://github.com/balena-io-modules/dockerfile-template
Commit: 3ae77d4ec75be1cb35b9abdab5ac54fc08443aa4

Change-type: minor

dfunckt commented:
Also added https://github.com/balena-io-modules/dockerfile-template

dfunckt commented:
@balena-ci I self-certify!
